### PR TITLE
Create new PRs. Fixes #22.

### LINKED ISSUE #22: Implement git-pr-create command
Create a new local branch with an associated upstream tracking branch for a pull request.

### PR DESCRIPTION
**Pull request by robertdfrench**
This pr builds out the `git-pr-create` command. I took a few big deviations from our previous discussions which I should call out:

* I went back to assuming "origin" is the one and only remote. Because `git pr-create` implies `git push`, we need to know the name of a remote that will receive the pushed objects. I decided to punt on solving this and go with "origin".
* `fake_git` still doesn't do a lot, but it's now **much** more convoluted. I think what is there works... but perhaps we should switch to clap.
* There is no integration test for `Git::push_upstream` (git push -u) because I wasn't sure how to tackle that

If you aren't comfortable punting on these, then we can address them before merging, but I think they'll each require a bit of brainstorming.

### CODE REVIEW

**Comment by robertdfrench** [comment]
@calebwherry I think you actually have to leave a review on this one. It's marked as auto-merge, but it's still blocked waiting on a review from you which does not contain any unresolved conversations.

**Comment by robertdfrench** [comment]
Oh! Or maybe you are waiting on review from me since you pushed last? I can't tell from this interface.

**Comment by calebwherry** [comment]
I was messing around in the app interface. I haven't reviewed it yet fully, too much to do on the app. My last push was the button to "get up to date with trunk" in the app, so I pushed it.

**Comment by robertdfrench** [comment]
Ah okay, good deal. Take your time, I'm getting some things done in parallel.

File: src/lib.rs
Comment (calebwherry) [inline src/lib.rs]: Refactor with new trunk functionality.

File: src/lib.rs
Comment (calebwherry) [inline src/lib.rs]: SAA

File: src/lib.rs
Comment (calebwherry) [reply src/lib.rs]: Fixed by e92e235.

File: src/lib.rs
Comment (calebwherry) [reply src/lib.rs]: Fixed by e92e235.

**Review by calebwherry — APPROVED** [review]

**Pull request merged**
